nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/security/ACE.java

```diff
@@ -69,11 +71,24 @@ public static ACE fromId(String aceId) {
             return null;
         }
 
-        String[] parts = aceId.split(":");
-        if (parts.length < 3) {
-            throw new IllegalArgumentException(String.format("Invalid ACE id: %s", aceId));
+	// An ACE is composed of tokens separated with ":" caracter
+	// First 3 tokens are mandatory; following 3 tokens are optional
```

Comment:
Also mention that the ":" separator is still present even if the tokens are absent, and that the first token (username) is allowed to contain embedded ":".